client/src/app/aai/aai.service.ts

```diff
@@ -0,0 +1,82 @@
+import {Profile, User, UserManager, UserManagerSettings, WebStorageStateStore} from 'oidc-client';
+import {Injectable} from '@angular/core';
+import {from, Observable} from 'rxjs';
+import {HttpClient} from '@angular/common/http';
+import {AlertService} from '../shared/services/alert.service';
+import {Router} from '@angular/router';
+import { environment } from '../../environments/environment';
+
+export function getClientSettings(): UserManagerSettings {
+  return {
+    authority: environment.AAI_AUTHORITY,
+    client_id: environment.AAI_CLIENT_ID,
+    redirect_uri: environment.AAI_REDIRECT_URI,
+    post_logout_redirect_uri: environment.AAI_POST_LOGOUT_REDIRECT_URI,
+    response_type: 'token id_token',
+    scope: 'email openid profile',
+    filterProtocolClaims: true,
+    loadUserInfo: true,
+    userStore: new WebStorageStateStore({store: window.localStorage})
+  };
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AaiService {
+
+  private manager = new UserManager(getClientSettings());
+  private user: User = null;
+
+  constructor(private http: HttpClient,
+              private alertService: AlertService,
+              private router: Router) {
+
+    this.getUser().then(user => {
+      this.user = user;
+    });
+  }
+
+  getUser(): Promise<User> {
+    return this.manager.getUser();
+  }
+
+  isLoggedIn(): Observable<boolean> {
+    return from(this.manager.getUser()).map(user => user && !this.user.expired);
+  }
+
+  isAuthenticated(): boolean {
+    return this.user && !this.user.expired;
```

Comment:
Some of these methods can throw null exceptions if the user hasn't yet been fetched from the UserManager. Could fix this by wrapping access to `this.user` in `getUser()` only, which returns a Observable<User> from manager.getUser() if `this.user` is null, or returns an `Observable.of(this.user)` it's not null. Might have to rename the existing `getUser()` then to something like fetch-user

Reply:
Applied comments.